heartbeat/mpathpersist

```diff
@@ -0,0 +1,684 @@
+OCF_RESKEY_master_score_delay=30
+
+#######################################################################
+# Initialization:
+
+: ${OCF_FUNCTIONS_DIR=${OCF_ROOT}/lib/heartbeat}
+. ${OCF_FUNCTIONS_DIR}/ocf-shellfuncs
+
+# set default values
+    : ${mpathpersist_binary=${OCF_RESKEY_binary_default}} 		# binary name for the resource
+    : ${devs=${OCF_RESKEY_devs_default}}                            	# device list
+    : ${required_devs_no=${OCF_RESKEY_required_devs_no_default}}   	# number of required devices
+    : ${reservation_type=${OCF_RESKEY_reservation_type}}		# reservation type
+    : ${master_score_base=${OCF_RESKEY_master_score_base}}		# master score base 
+    : ${master_score_dev_factor=${OCF_RESKEY_master_score_dev_factor}}	# device factor for master score
+    : ${master_score_delay=${OCF_RESKEY_master_score_delay}}		# delay for master score
+
+#######################################################################
+
+
+meta_data() {
+    cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="mpathpersist">
+<version>1.1</version>
+
+<longdesc lang="en">
+This resource agent manages SCSI persistent reservations on multipath devices.
+"mpathpersist" from multipath-tools is used, please see its documentation.
+Should be used as multistate (Master/Slave) resource
+Slave registers its node id ("crm_node -i") as reservation key ( --param-sark ) on each device in the params "devs" list.
+Master reservs all devices from params "devs" list with reservation "--prout-type" value from "reservation_type" parameter.
+Please see man sg_persist(8) and mpathpersist(8) for reservation_type details.
+</longdesc>
+<shortdesc lang="en">Manages SCSI persistent reservations on multipath devices</shortdesc>
+
+<parameters>
+<parameter name="binary" unique="0">
+<longdesc lang="en">
+The name of the binary that manages the resource.
+</longdesc>
+<shortdesc>the binary name of the resource</shortdesc>
+<content type="string" default="$mpathpersist_binary"/>
+</parameter>
+
+<parameter name="devs" unique="0" required="1">
+<longdesc lang="en">
+Device list. Multiple devices can be listed with blank space as separator. 
+Shell wildcars are allowed.
+</longdesc>
+<shortdesc lang="en">device list</shortdesc>
+<content type="string"/>
+</parameter>
+
+<parameter name="required_devs_no" unique="0" required="0">
+<longdesc lang="en">
+Minimum number of "working" devices from device list
+      1) existing 
+      2) "mpathpersist --in --read-keys \$device" works (Return code 0)
+resource actions "start","monitor","promote" and "validate-all" return "\$OCF_ERR_INSTALLED"
+if the actual number of "working" devices is less then "required_devs_no".
+resource actions "stop" and "demote" tries to remove reservations and registration keys from 
+all working devices, but always return "\$OCF_SUCCESS"
+</longdesc>
+<shortdesc lang="en">minimum number of working devices</shortdesc>
+<content type="string" default="1"/>
```

Comment:
Should also use `OCF_RESKEY_required_devs_no_default` here (same for other parameters)